src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -297,11 +312,12 @@ def acr_task_update(cmd,  # pylint: disable=too-many-locals
                             token_type=DEFAULT_TOKEN_TYPE
                         )
                     ),
-                    source_trigger_events=[SourceTriggerEvent.commit],
+                    source_trigger_events=_events if (commit_trigger_enabled or pull_request_trigger_enabled) else None,
```

Comment:
We can end up setting `source_trigger_events` to None if both `commit_trigger_enabled` and `pull_request_trigger_enabled` are `False` (say user trying to update just the `status`) but there was an existing trigger with events?

Reply:
This None set is here to hit an optimization in the RP that checks if the user is updating just the status of a trigger. (This optimization expects SourceTriggerEvents to be none). 
```
        public bool IsOnlyStatusUpdate() =>
            !string.IsNullOrWhiteSpace(Status) &&
            SourceTriggerEvents?.Any() != true &&
            SourceRepository == null;
```

In previous designs, we were never hitting this optimization since we were always sending a payload with an event in the source_trigger_events parameter. 

The RP prevents a task update from removing all events from an existing trigger:
```
            sourceTrigger.SourceTriggerEvents = sourceTrigger.SourceTriggerEvents?.Count() > 0 ?
                sourceTrigger.SourceTriggerEvents : 
                existingTrigger.SourceTriggerEvents;
```